driver/tests/mpi/test_restart.py

```diff
@@ -0,0 +1,188 @@
+            "qliquid",
+            "qice",
+            "qrain",
+            "qsnow",
+            "qgraupel",
+        ]:
+            np.testing.assert_allclose(
+                restart[var].isel(time=0).values, regular[var].isel(time=-1).values
+            )
+
+        # now use the same restart to test DriverState load function
+        with open("RESTART/restart.yaml", "r") as f:
+            config = yaml.safe_load(f)
+            config["comm_config"]["type"] = "null_comm"
+            config["comm_config"]["config"]["rank"] = 0
+            config["comm_config"]["config"]["total_ranks"] = 6
+            driver_config = DriverConfig.from_dict(config)
+        driver_state = DriverState.load_state_from_restart("RESTART", driver_config)
+        assert isinstance(driver_state, DriverState)
+
+        restart_dycore = xr.open_dataset(
+            f"RESTART/restart_dycore_state_{driver_config.comm_config.config.rank}.nc"
+        )
+        for var in driver_state.dycore_state.__dict__.keys():
+            if isinstance(driver_state.dycore_state.__dict__[var], pace.util.Quantity):
+                np.testing.assert_allclose(
+                    driver_state.dycore_state.__dict__[var].data,
+                    restart_dycore[var].values,
+                )
+
+        restart_physics = xr.open_dataset(
+            f"RESTART/restart_physics_state_{driver_config.comm_config.config.rank}.nc"
+        )
+        for var in driver_state.physics_state.__dict__.keys():
+            if isinstance(
+                driver_state.physics_state.__dict__[var],
+                gt4py.storage.storage.CPUStorage,
+            ):
+                np.testing.assert_allclose(
+                    driver_state.physics_state.__dict__[var].data,
+                    restart_physics[var].values,
+                )
+    finally:
+        subprocess.Popen(["make", "clean"])
+
+
+def test_restart_save_to_disk():
+    try:
+        backend = "gtc:numpy"
+        mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
+        partitioner = pace.util.CubedSpherePartitioner(
+            pace.util.TilePartitioner((1, 1))
+        )
+        communicator = pace.util.CubedSphereCommunicator(mpi_comm, partitioner)
+        sizer = pace.util.SubtileGridSizer.from_tile_params(
+            nx_tile=12,
+            ny_tile=12,
+            nz=79,
+            n_halo=3,
+            extra_dim_lengths={},
+            layout=(1, 1),
+            tile_partitioner=partitioner.tile,
+            tile_rank=communicator.tile.rank,
+        )
+        quantity_factory = pace.util.QuantityFactory.from_backend(
+            sizer=sizer, backend=backend
+        )
+
+        init = BaroclinicConfig()
+        driver_state = init.get_driver_state(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+        driver_state.save_state(mpi_comm)
+
+        restart_dycore = xr.open_dataset(
+            f"RESTART/restart_dycore_state_{mpi_comm.rank}.nc"
```

Comment:
this will only compare rank 0, is that intended?

Reply:
Yes since it runs serially I think it's sufficient if we just test rank 0.